ppx.ml

```diff
@@ -0,0 +1,133 @@
+module Stre = Str
+open Parsetree
+open Ast_helper
+module AM = Ast_mapper
+
+let error ppf =
+  Format.kasprintf (fun s ->
+    let loc = !default_loc in
+    Exp.extension ~loc @@
+    AM.extension_of_error (Location.error ~loc s))
+    ppf
+
+let mklid m lid =
+  Exp.ident
+    {Location.loc = !default_loc ; txt = Longident.(Ldot (Lident m, lid))}
+let app m lid l =
+  Exp.(apply (mklid m lid)
+      (List.map (fun e -> Asttypes.Nolabel, e) l))
+
+let try_int s =
+  try `Int32 (Int32.of_string s)
```

Comment:
While implementing this, I realized there is a bug in the implementation:
```ocaml
# 0x8fffffffl ;;
- : int32 = -1879048193l
# 0x8fffffffz ;;
- : Z.t = -1879048193
```

Which is a bit unfortunate. I'm not sure yet what's the best way to solve it. Until know, I tried to avoid using zarith inside the ppx itself but maybe It would be best, to avoid this kind of issues ...